de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/dvlm/mat/Exporter.java

```diff
@@ -0,0 +1,290 @@
+				return contentOfProperty(object);
+			}
+
+		}.doSwitch(element);
+		return propertyArray;
+	}
+
+	//create a struct for the fitting PropertyInstance
+	/**
+	 * creates a new .mat-struct with all information about UnitValuePropertyInstance
+	 * only unit and value are saved
+	 * @param element UnitValuePropertyInstance
+	 */
+	private Array contentOfProperty(UnitValuePropertyInstance element) {
+		Struct struct = Mat5.newStruct();
+
+		if (element.getType() instanceof FloatProperty) {
+			BeanPropertyFloat bpf = new BeanPropertyFloat(element);
+			struct.set("unit", Mat5.newString(bpf.getUnit()));
+			struct.set("value", (bpf.getValue() == Double.NaN) ? Mat5.newString("") : Mat5.newScalar(bpf.getValue()));
+		} else if (element.getType() instanceof IntProperty) {
+			BeanPropertyInt bpi = new BeanPropertyInt(element);
+			struct.set("unit", Mat5.newString(bpi.getUnit()));
+			struct.set("value", (!bpi.isSet()) ? Mat5.newString("") : Mat5.newScalar(bpi.getValue()));
+		}
+		return struct;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about ValuePropertyInstance
+	 * only Value is saved
+	 * @param element ValuePropertyInstance
+	 */
+	private Struct contentOfProperty(ValuePropertyInstance element) {
+		Struct struct = Mat5.newStruct();
+		if (element.getType() instanceof BooleanProperty) {
+			BeanPropertyBoolean bpb = new BeanPropertyBoolean(element);
+			struct.set("value", (!bpb.isSet()) ? Mat5.newString("") : Mat5.newLogicalScalar(bpb.getValue()));
+		} else if (element.getType() instanceof StringProperty) {
+			BeanPropertyString bps = new BeanPropertyString(element);
+			struct.set("value", (!bps.isSet()) ? Mat5.newString("") : Mat5.newString(bps.getValue()));
+		}
+		return struct;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about EReferencePropertyInstance
+	 * only uuid, reference and reference-class are saved
+	 * @param element EReferencePropertyInstance
+	 */
+	private Array contentOfProperty(EReferencePropertyInstance element) {
+		Struct struct = Mat5.newStruct();
+		if (element.getType() instanceof EReferenceProperty) {
+			
+			BeanPropertyEReference<EReferenceProperty> bpe = new BeanPropertyEReference<EReferenceProperty>(element);
+			
+			struct.set("reference", (!bpe.isSet()) ? Mat5.newString("") : Mat5.newString(bpe.getValue().toString()));
+			struct.set("reference-class", (!bpe.isSet()) ? Mat5.newString("") : Mat5.newString(bpe.getValue().getClass().getName()));
+		}
+		return struct;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about ReferencePropertyInstance
+	 * only uuid and fullQualifiedInstanceName are saved
+	 * @param element ReferencePropertyInstance
+	 */
+	private Array contentOfProperty(ReferencePropertyInstance element) {
+		Struct struct = Mat5.newStruct();
+		if (element.getType() instanceof ReferenceProperty) {
+			if (element.getReference() != null) {
+				ATypeInstance referencedTypeInstance = element.getReference();

```

Comment:
We can move this before the if check and then check `if (referencedTypeInstance != null) `